src/main/java/uk/gov/hmcts/darts/event/service/impl/EventServiceImpl.java

```diff
@@ -19,11 +23,26 @@ public class EventServiceImpl implements EventService {
     private final EventMapper eventMapper;
     private final EventIdValidator eventIdValidator;
     private final EventRepository eventRepository;
+    private final DataAnonymisationService dataAnonymisationService;
 
     @Override
     public AdminGetEventForIdResponseResult adminGetEventById(Integer eventId) {
         eventIdValidator.validate(eventId);
         Optional<EventEntity> eventEntityOptional = eventRepository.findById(eventId);
         return eventMapper.mapToAdminGetEventsResponseForId(eventEntityOptional);
     }
+
+    @Override
+    @Transactional
+    public void adminObfuscateEveByIds(List<Integer> eveIds) {
+        eveIds.stream()
+            .map(this::getEventEntityById)
+            .distinct()
+            .forEach(dataAnonymisationService::anonymizeEvent);
+    }
+
+    EventEntity getEventEntityById(Integer eveId) {
+        return eventRepository.findById(eveId)
```

Comment:
The requirements seem to talk about:-

When the event_id is zero for the relevant eve_id needs to be obfuscated

Reply:
So the eve_id zero comes from a old requirement which has since been removed around obfuscating previous versions of events but this has been moved to a new ticket now.

This ticket exclusively obfuscates by eve_id

Reply:
oh ok, thats fine. Can we mention that on the ticket if its not already called out

Reply:
@jackmaloney see above